doc/user-guide.adoc

```diff
@@ -315,6 +318,66 @@ spec:
         name: self-signed
     termination: reencrypt
 ----
+==== Using automatic registration with OIDC providers
+
+The operator can request a client Id and client Secret from providers, rather than requiring them in advance. This can simplify deployment, as the provider's administrator can supply the information needed for registration once, instead of supplying clientIds and secrets repetitively.  An additional secret specified by `sso.oidc[].autoRegisterSecret` contains additional information necessary to perform the registration.  First the operator will make an https request to the `sso.oidc[].discoveryEndpoint` to obtain URLs for subsequent REST calls.  Next it will make additional REST calls to the provider and obtain a client Id and client Secret. Automatic registration takes precedence over the values in the sso secret. The name of this secret should end with `-olapp-sso` for it to be monitored for changes.  This is tested with Red Hat Single Sign-on (RHSSO). 
```

Comment:
"_The name of this secret should end with `-olapp-sso` for it to be monitored for changes._"
With the current implementation you have in place, I don't think the auto-register secret is monitored.

Instead of users always having to explicitly specify the auto-register secret name, by default we could automatically find it (only while reconciling providers) using a naming convention: <CR_name>-olapp-sso-<provider-id>

Reply:
I will look into auto-detecting off of the name, to make it consistent with the way we handle the other sso secret. I can't think of any disadvantage to that.